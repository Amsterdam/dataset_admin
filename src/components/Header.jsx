import React from 'react';
import { NavLink } from 'react-router-dom';
import { Container, Menu } from 'semantic-ui-react';

import Logo from '../styling/logo.svg';

import { logout } from '../services/auth/auth';

const Header = () => (
  <Menu fixed="top">
    <Container>
      <Logo />
      <Menu.Item header>
        <NavLink to="/dcatd_admin/datasets">
          DataPunt catalogus beheer
        </NavLink>
      </Menu.Item>
      <Menu.Item>
        <NavLink to="/dcatd_admin/datasets">
          Datasets
        </NavLink>
      </Menu.Item>
      <Menu.Menu position="right">
        <Menu.Item onClick={logout}>
          Uitloggen
        </Menu.Item>
      </Menu.Menu>
    </Container>
  </Menu>
);

export default Header;
