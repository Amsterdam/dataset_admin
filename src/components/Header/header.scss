@import "../../styling/config.scss";

.header {
  box-shadow: 0 2px 0 0 $warm-grey;
  display: flex;
  height: 50px;

  .logo,
  .title {
    align-self: center;
  }

  .logo {
    flex-basis: 87px;
  }
}
