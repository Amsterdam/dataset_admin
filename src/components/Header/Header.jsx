import React from 'react';
import { Container, Header as SemanticHeader } from 'semantic-ui-react';

import { logout } from '../../services/auth/auth';

import Logo from '../../styling/logo.svg';

import './header.scss';

const Header = () => (
  <Container>
    <SemanticHeader size="small" className="header">
      <Logo className="logo" />
      <span className="title">Dataset beheer</span>
      {window.sessionStorage.getItem('DCATD_DETAIL_REDIRECT_URL') || window.sessionStorage.getItem('DCATD_LIST_REDIRECT_URL') ?
        '' :
        <button
          onClick={logout}
        >uitloggen</button>
      }
    </SemanticHeader>
  </Container>
);

export default Header;
