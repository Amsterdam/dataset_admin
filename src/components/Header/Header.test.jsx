import React from 'react';

import Header from './Header';

describe('The Header component', () => {
  let mockUrl = 'http://portal/redirect-url';
  let wrap;
  let origSessionStorage;

  beforeEach(() => {
    origSessionStorage = window.sessionStorage;

    window.sessionStorage = {
      getItem: () => mockUrl
    };
  });

  afterEach(() => {
    window.sessionStorage = origSessionStorage;
  });

  it('renders component with DCATD_LIST_REDIRECT_URL', () => {
    wrap = shallow(<Header />);

    expect(wrap).toMatchSnapshot();
  });

  it('renders component without DCATD_LIST_REDIRECT_URL', () => {
    mockUrl = null;

    wrap = shallow(<Header />);

    expect(wrap).toMatchSnapshot();
  });
});
