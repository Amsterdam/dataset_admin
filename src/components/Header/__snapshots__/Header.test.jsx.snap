// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders component with DCATD_LIST_REDIRECT_URL 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Header
        className="header"
        size="small"
      >
        <a
          className="logo-link"
          href="http://portal/redirect-url"
        >
          <logo.svg
            className="logo"
          />
        </a>
        <span
          className="title"
        >
          Dataset beheer
        </span>
        
      </Header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <a
            className="logo-link"
            href="http://portal/redirect-url"
          >
            <logo.svg
              className="logo"
            />
          </a>,
          <span
            className="title"
          >
            Dataset beheer
          </span>,
          "",
        ],
        "className": "header",
        "size": "small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <logo.svg
              className="logo"
            />,
            "className": "logo-link",
            "href": "http://portal/redirect-url",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
            },
            "ref": null,
            "rendered": null,
            "type": "logo.svg",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dataset beheer",
            "className": "title",
          },
          "ref": null,
          "rendered": "Dataset beheer",
          "type": "span",
        },
        "",
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Header
          className="header"
          size="small"
        >
          <a
            className="logo-link"
            href="http://portal/redirect-url"
          >
            <logo.svg
              className="logo"
            />
          </a>
          <span
            className="title"
          >
            Dataset beheer
          </span>
          
        </Header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <a
              className="logo-link"
              href="http://portal/redirect-url"
            >
              <logo.svg
                className="logo"
              />
            </a>,
            <span
              className="title"
            >
              Dataset beheer
            </span>,
            "",
          ],
          "className": "header",
          "size": "small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <logo.svg
                className="logo"
              />,
              "className": "logo-link",
              "href": "http://portal/redirect-url",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logo",
              },
              "ref": null,
              "rendered": null,
              "type": "logo.svg",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dataset beheer",
              "className": "title",
            },
            "ref": null,
            "rendered": "Dataset beheer",
            "type": "span",
          },
          "",
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The Header component renders component without DCATD_LIST_REDIRECT_URL 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Header
        className="header"
        size="small"
      >
        <logo.svg
          className="logo"
        />
        <span
          className="title"
        >
          Dataset beheer
        </span>
        <button
          onClick={[Function]}
        >
          uitloggen
        </button>
      </Header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <logo.svg
            className="logo"
          />,
          <span
            className="title"
          >
            Dataset beheer
          </span>,
          <button
            onClick={[Function]}
          >
            uitloggen
          </button>,
        ],
        "className": "header",
        "size": "small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": "logo.svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dataset beheer",
            "className": "title",
          },
          "ref": null,
          "rendered": "Dataset beheer",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "uitloggen",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "uitloggen",
          "type": "button",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Header
          className="header"
          size="small"
        >
          <logo.svg
            className="logo"
          />
          <span
            className="title"
          >
            Dataset beheer
          </span>
          <button
            onClick={[Function]}
          >
            uitloggen
          </button>
        </Header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <logo.svg
              className="logo"
            />,
            <span
              className="title"
            >
              Dataset beheer
            </span>,
            <button
              onClick={[Function]}
            >
              uitloggen
            </button>,
          ],
          "className": "header",
          "size": "small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
            },
            "ref": null,
            "rendered": null,
            "type": "logo.svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dataset beheer",
              "className": "title",
            },
            "ref": null,
            "rendered": "Dataset beheer",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "uitloggen",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "uitloggen",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
